cloud-core/src/main/java/cloud/commandframework/CommandTree.java

```diff
@@ -449,87 +449,7 @@ private CommandTree(final @NonNull CommandManager<C> commandManager) {
-                            String.format("%s_%d", child.getValue().getName(), i),
-                            commandQueue.remove()
-                    );
-                }
-            }
-
-            if (child.getValue() != null) {
-                if (commandQueue.isEmpty()) {
-                    return Collections.emptyList();
-                } else if (child.isLeaf() && commandQueue.size() < 2) {
-                    commandContext.setCurrentArgument(child.getValue());
-                    return child.getValue().getSuggestionsProvider().apply(commandContext, commandQueue.peek());
-                } else if (child.isLeaf()) {
-                    if (child.getValue() instanceof CompoundArgument) {
-                        final String last = ((LinkedList<String>) commandQueue).getLast();
-                        commandContext.setCurrentArgument(child.getValue());
-                        return child.getValue().getSuggestionsProvider().apply(commandContext, last);
-                    }
-                    return Collections.emptyList();
-                } else if (commandQueue.peek().isEmpty()) {
-                    commandContext.setCurrentArgument(child.getValue());
-                    return child.getValue().getSuggestionsProvider().apply(commandContext, commandQueue.remove());
-                }
-
-                // START: Preprocessing
-                final ArgumentParseResult<Boolean> preParseResult = child.getValue().preprocess(
-                        commandContext,
-                        commandQueue
-                );
-                if (preParseResult.getFailure().isPresent() || !preParseResult.getParsedValue().orElse(false)) {
-                    final String value = commandQueue.peek() == null ? "" : commandQueue.peek();
-                    commandContext.setCurrentArgument(child.getValue());
-                    return child.getValue().getSuggestionsProvider().apply(commandContext, value);
-                }
-                // END: Preprocessing
-
-                // START: Parsing
-                commandContext.setCurrentArgument(child.getValue());
-                final ArgumentParseResult<?> result = child.getValue().getParser().parse(commandContext, commandQueue);
-                if (result.getParsedValue().isPresent()) {
-                    commandContext.store(child.getValue().getName(), result.getParsedValue().get());
-                    return this.getSuggestions(commandContext, commandQueue, child);
-                } else if (result.getFailure().isPresent()) {
-                    final String value = commandQueue.peek() == null ? "" : commandQueue.peek();
-                    commandContext.setCurrentArgument(child.getValue());
-                    return child.getValue().getSuggestionsProvider().apply(commandContext, value);
-                }
-                // END: Parsing
-            }
+            return suggestionsForDynamicArgument(commandContext, commandQueue,  children.get(0));
```

Comment:
```suggestion
            return this.suggestionsForDynamicArgument(commandContext, commandQueue,  children.get(0));
```

Reply:
Changed that, and removed the double-space before children.get